pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -207,15 +309,31 @@ func (c *objectCache) Get(namespace, name string) (runtime.Object, error) {
 		// - doing that would require significant refactoring to reflector
 		// we limit ourselves to just quickly stop the reflector here.
 		if c.isImmutable(object) {
+			item.markImmutable()
 			if item.stop() {
 				klog.V(4).Infof("Stopped watching for changes of %q/%q - object is immutable", namespace, name)
 			}
 		}
+
+		item.updateLastAccessTime(c.clock.Now())
```

Comment:
This should be called unconditionally (even if the object doesn't exist in cache).
So probably somewhere around like 291.